pkg/util/validation/validate.go

```diff
@@ -82,7 +82,7 @@ func DiscardedSamplesCounter(reg prometheus.Registerer, reason string) *promethe
 		ConstLabels: map[string]string{
 			discardReasonLabel: reason,
 		},
-	}, []string{"user"})
+	}, []string{"user", "custom_user_label"})
```

Comment:
I'm not a big fan of `"custom_user_label"` label name. In the PR http://github.com/grafana/mimir/pull/2702 it was called "group" which I think makes more sense, because we want to provide a way to track incoming series by a custom grouping.